custom-metrics-stackdriver-adapter/pkg/adapter/translator/response_translator.go

```diff
@@ -243,6 +248,11 @@ func (t *Translator) metricKey(timeSeries *stackdriver.TimeSeries) (string, erro
 		case "k8s_node":
 			return ":" + timeSeries.Resource.Labels["node_name"], nil
 		case "prometheus_target":
+			if resourceSchema == "nodes" {
+				// for prometheus_target metric, its instance usually consist of node id and the port who emited the metric.
```

Comment:
I think we want to use the node label from the metrics instead of the instance. For example, a kube-state-metrics metrics exposes `kube_node_status_condition` (https://github.com/kubernetes/kube-state-metrics/blob/master/docs/node-metrics.md), the instance will be the node that KSM is deployed. But we want to use the node label exported from the metric.

Reply:
an example prometheus metric could be like this
```json
{ 
     "metric": {
        "labels": {
          "pod": "node-exporter-lfx95"
        },
        "type": "prometheus.googleapis.com/node_load1/gauge"
      },
      "resource": {
        "type": "prometheus_target",
        "labels": {
          "location": "",
          "project_id": "",
          "job": "node-exporter",
          "instance": "gke-stackdriver-adapter--default-pool-cee13989-0i75:8080",
          "cluster": "stackdriver-adapter-dev",
          "namespace": ""
        }
}
```
there is not always a label called `node`, or I misunderstood

Reply:
for some metrics, there will be a node label, like `kube_node_status_condition` (You can follow https://cloud.google.com/stackdriver/docs/managed-prometheus/exporters/kube_state_metrics to try kube-state-metrics metrics). For some metrics from node-exporter, as your example, the node name can be inferred from the instance. 

Can we do something like this, if there is a node metric label, we use the node value; otherwise, we use the instance value?